frontend/src/pages/PipelineDetails.tsx

```diff
@@ -369,8 +388,14 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
       this.props.history.replace({
         pathname: `/pipelines/details/${this.state.pipeline.id}/version/${versionId}`,
       });
+      const graph = await this._createGraph(selectedVersionPipelineTemplate);
+      let reducedGraph = graph ? transitiveReduction(graph) : undefined;
```

Comment:
Can we use transitiveReduction(graph) as a way to return undefined if graph doesn't exist? Then it will look like 

```
let reducedGraph = transitiveReduction(graph);
```

Reply:
For discussion: personally I find current code to be more readable. If we move undefined handling inside the function, it's not clear when graph will be undefined on both places. It's not clear here, because it's a simple function call. It's also not clear in the function, because it can have many call sites.

Reply:
@zijianjoy We could, but `graph` can be `null`, not `undefined`. This is the signature of `_createGraph`: `private async _createGraph(templateString: string): Promise<dagre.graphlib.Graph | null>`.

Reply:
We would have to change signature of `transitiveReduction` to `transitiveReduction(graph: dagre.graphlib.Graph | null): dagre.graphlib.Graph | undefined`, which isn't that clean IMHO

Reply:
(@Bobgy sorry my page hadn't refreshed so I wrote my comments without seeing yours first. But we are agreeing :) )

Reply:
Thank you both for clarification! That makes sense in terms of restricting the input parameter to be not undefined.